luigi/task.py

```diff
@@ -575,6 +575,40 @@ def output(self):
         """
         return []  # default impl
 
+    def process_requires(self, requires):
+        """
+        Override in "template" tasks which themselves are supposed to be
+        subclassed and thus have their requires() overridden (name preserved to
+        provide consistent end-user experience), yet need to introduce
+        custom behaviour into dependencies.
+
+        Differences with :py:meth:`_requires`:
```

Comment:
We shouldn't mention the internal function should we?

Reply:
Documentation for `_requires` demonstrates similar use case(Advanced use with template tasks). So I think it's okay.

Reply:
Can you add to the doc-string for `_requires()` (and also for `deps()`) and just mention "DEPRECATED don't override". For the `_requires()`-use-cases I would it if `process_requires()` could be the new go-to-replacement for it.